packages/features/tasker/tasks/executeAIPhoneCall.ts

```diff
@@ -107,15 +114,78 @@ export async function executeAIPhoneCall(payload: string) {
       });
 
       if (!hasCredits) {
-        log.warn(`Insufficient credits for AI phone call`, {
+        log.warn(`Insufficient credits for AI phone call for workflow reminder ${data.workflowReminderId}`, {
           userId: data.userId,
           teamId: data.teamId,
           workflowReminderId: data.workflowReminderId,
           bookingUid: workflowReminder.booking?.uid,
         });
-        throw new Error(
-          `Insufficient credits to make AI phone call. Please purchase more credits. user: ${data?.userId}, team: ${data?.teamId}`
-        );
+
+        try {
+          let teamWithAdmins:
+            | {
+                id: number;
+                name: string;
+                adminAndOwners: { id: number; name: string | null; email: string; t: TFunction }[];
+              }
+            | undefined;
+          let user:
+            | {
+                id: number;
+                name: string | null;
+                email: string;
+                t: TFunction;
+              }
+            | undefined;
+
+          if (data.teamId) {
+            const teamRepository = new TeamRepository(prisma);
+            const team = await teamRepository.findTeamWithAdminMembers({ teamId: data.teamId });
+
+            if (team) {
+              teamWithAdmins = {
+                id: team.id,
+                name: team.name ?? "",
+                adminAndOwners: await Promise.all(
+                  team.members.map(async (member) => ({
+                    id: member.user.id,
+                    name: member.user.name,
+                    email: member.user.email,
+                    t: await getTranslation(member.user.locale ?? "en", "common"),
+                  }))
+                ),
+              };
+            }
+          } else if (data.userId) {
+            const userRepository = new UserRepository(prisma);
+            const userRecord = await userRepository.findById({ id: data.userId });
+
+            if (userRecord) {
+              user = {
```

Comment:
This code could benefit from abstraction. Something like _getUserAndTeam_

Reply:
infact could be better to just abstract entire logi into separate handler fn like handleInsufficientCredits.